bip-hashrate-escrows.mediawiki

```diff
@@ -0,0 +1,469 @@
+# A "blinded TxID" is way of hashing the txn, in which we first overwrite some parts of the txn with zeros. Specifically, the first 36 bytes of "TxIn0" (the first input, including TxOutHash and TxOutIndex), as well as the first 8 bytes of "TxOut0" (the first output).
+
+Blinding is necessary because we allow each sidechain only one UTXO at a time.
+
+of our restriction of the account to a single UTXO-member. Because of this, during the ACKing process the withdrawal-txn (which is currently being ACKed) may change in two ways: the CTIP (which changes with each deposit), and the total quantity of BTC stored in the account (which arbitrarily increases with each new deposit). In other words, a withdrawal-attempt is created via M3, but this takes place many blocks before the withdrawal is actually included via M6. During this time, a single new deposit to the account would change its CTIP and its value. So, what do we ACK? Well, we ACK a "blinded" version of the withdrawal. This blinded version is stable because the dynamic parts are always overwritten with zeros.
+
+While we ACK a blinded WT^, what is actually included in the blockchain ("M6") is an unblinded WT^. Since each blinded WT^ could correspond to many different unblinded WT^s, we need to impose further restrictions on those unblinded WT^s that are finally included. First, we will force the final unblinded WT^ to spend the entire sidechain balance (by forcing sum(input_values) to equal sum(output_values)). To avoid withdrawing the entire sidechain balance with every withdrawal, we will, secondly, force the unblinded WT^ to create a new output which is itself a deposit to the sidechain it withdrew from (which nodes can check using D1's CTIP field). Unfortunately, these requirements eliminate the possibility of including a transaction fee, as traditionally calculated. So, finally, to compensate for *that*, txn fees are encoded explicitly as a withdrawal to OP_TRUE (which the main:block's miner can immediately claim).
+
+With all of this in place, the only requirements for inclusion in a block are these:
+
+# "Be ACKed" -- The "blinded TxID" of this txn must be member of the "approved candidate" set in the D2 of this block.
+# "Return Change to Account" -- TxOut0 must pay to the "critical account" (see D1) that corresponds to the CTIP that was selected as a TxIn.
+# "Return *all* Change to Account" -- Sum of inputs must equal the sum of outputs. No traditional tx fee is possible.
+
+Finally, don't forget that M6 inherits the requirement (common to both M5 and M6) that the CTIP be selected as an input, and that the CTIP then be updated. In this case, we know that the critical index will be zero, so the new CTIP will be ("this TxID" (NOT blinded), 0). The TxID is NOT blinded because blinding is only for accumulating ACKs.
+
+As a result of these requirements, every single withdrawal-attempt will fail, unless an entry has been added to D2 and "ACKed" a sufficient number of times.
+
```

Comment:
The above seems to be specifying extension blocks, not sidechains, which is a whole can of worms, and *should not* be considered acceptable to be done by miners at will.

Reply:
In this proposal, since the sidechains are so asymmetric, a sidechain is considered to be equivalent to an optional extension block.

And since they are optional, it makes no difference if anyone does them at will.

And since one purpose of these sidechains is "permissionless innovation" (ie, to provide a way for Bitcoin to recover from errors in the dev process / BIP process), it is actually imperative that there be some way of adding new sidechains even if the BIP process doesn't want them.

Reply:
But if a miner doesn't do a sidechain, won't it be interpreted as a NACK on every withdrawal?

Reply:
No, a sidechain can "abstain" and take no position on the withdrawals. In that case it is neither and ACK nor a NACK.

In practice, abstaining miners may also choose to rely on much...flimsier methods, like just checking in with someone who runs a sidechain node in real life, and asking them if everything is operating normally.

(In any well-designed sidechain, the withdrawal-IDs would be right in *each* sidechain header, or would else be very very easy to find. And there should only be 1 or 2 withdrawls there, per 3 months. And each of these should be a mere 32-bytes of information).

Reply:
Let me add that if more than 50% of the hashpower abstains completely (refusing to ACK or NACK), then no withdrawals will ever succeed. That is a function of the parameter choices of 13,150 and 26,300. We could make 26,300 longer, I suppose.

In fact, perhaps a good tradeoff to your other point would be to make *this* value flexible. Ie, keep the 13,150 required ACK score fixed, but allow the withdrawal period to be even longer.